src/api/objects/ObjectAPI.js

```diff
@@ -215,6 +227,188 @@ define([
+                let childObjects = {};
+                let promises = Object.keys(domainObject.configuration.layout.panels).map(key => {
+                    return openmct.objects.get(key)
+                        .then(object => {
+                            childObjects[key] = object;
+                        });
+                });
+
+                return Promise.all(promises)
+                    .then(function () {
+                        return migrateDisplayLayout(domainObject, childObjects);
+                    });
+            },
+        },
+        {
+            check(domainObject) {
+                return domainObject.type === 'telemetry.fixed' && domainObject.configuration['fixed-display'];
+            },
+            migrate(domainObject) {
+                const DEFAULT_GRID_SIZE = [64, 16];
+                let newLayoutObject = {
+                    identifier: domainObject.identifier,
+                    location: domainObject.location,
+                    name: domainObject.name,
+                    type: "layout"
+                };
+                let layoutType = openmct.types.get('layout');
+                layoutType.definition.initialize(newLayoutObject);
+                newLayoutObject.composition = domainObject.composition;
+                newLayoutObject.configuration.layoutGrid = domainObject.layoutGrid || DEFAULT_GRID_SIZE;
+
+                let elements = domainObject.configuration['fixed-display'].elements;
+                let telemetryObjects = {};
+                let promises = elements.map(element => {
+                    if (element.id) {
+                        return openmct.objects.get(element.id)
+                            .then(object => {
+                                telemetryObjects[element.id] = object;
+                            });
+                    }
+                });
+
+                return Promise.all(promises)
+                    .then(function () {
+                        newLayoutObject.configuration.items =
+                            migrateFixedPositionConfigurataion(elements, telemetryObjects);
+                        return newLayoutObject;
+                    });
+            }
+        }
+    ];
+
+    function needsMigration(domainObject) {
+        return migrations.some(m => m.check(domainObject));
+    }
+
+    function migrateObject(domainObject) {
+        return migrations.filter(m => m.check(domainObject))
+            .reduce((o, m) => "").migrate(domainObject);
+    }
+
+    function isTelemetry(domainObject) {
+        if (openmct.telemetry.isTelemetryObject(domainObject)
+            && domainObject.type !== 'summary-widget'
+            && domainObject.type !== 'example.imagery') {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    function migrateDisplayLayout(domainObject, childObjects) {
+        const DEFAULT_GRID_SIZE = [32, 32];
+        let migratedObject = {...domainObject};
```

Comment:
The spread operator for object literals is unfortunately not supported by all browsers yet. Maybe we could use `Object.assign({}, domainObject)` instead?

Reply:
There are other places in the code base where the spread operator is used. But that's a fair point. I'll change it to Object.assign().